java/client/src/main/java/glide/utils/ArrayTransformUtils.java

```diff
@@ -75,6 +75,27 @@ public static <T, U extends T> U[] castArray(T[] objectArr, Class<U> clazz) {
                 .toArray(size -> (U[]) Array.newInstance(clazz, size));
     }
 
+    /**
+     * Casts an <code>Object[][]</code> to <code>T[][]</code> by casting each nested array and every
+     * array element.
+     *
+     * @param outerObjectArr Array of arrays of objects to cast.
+     * @param clazz The class of the array elements to cast to.
+     * @return An array of arrays of type U, containing the elements from the input array.
+     * @param <T> The base type from which the elements are being cast.
+     * @param <U> The subtype of T to which the elements are cast.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T, U extends T> U[][] castArrayofArrays(T[] outerObjectArr, Class<U> clazz) {
+        if (outerObjectArr == null) {
+            return null;
+        }
+        for (int i = 0; i < outerObjectArr.length; i++) {
+            outerObjectArr[i] = (T) castArray((T[]) outerObjectArr[i], clazz);
```

Comment:
create a new array to convert values, rather than the given array.  This could cause unexpected behaviour is the caller isn't aware that the given array is being mutated.